### PR TITLE
Change the pattern of generic class metadata instantiation

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

File: stdlib/public/runtime/Metadata.cpp [hunk 1]
```diff
@@ -142,65 +226,148 @@ static GenericMetadataCache &unsafeGetInitializedCache(
   return lazyCache->unsafeGetAlreadyInitialized();
 }
 
+#ifndef NDEBUG
```
Comment (huonw) [inline stdlib/public/runtime/Metadata.cpp]: It looks like this conditional doesn't match the uses of `_objc_empty_cache` (or vice versa). https://ci.swift.org/job/oss-swift_tools-RA_stdlib-RD_test-simulator/714 rdar://problem/38132762

Comment (rjmccall) [reply stdlib/public/runtime/Metadata.cpp]: Yes, that was noticed yesterday and fixed in #14970.

**Pull request merged**